extension/otto/gohan_util.go

```diff
@@ -223,14 +274,39 @@ func init() {
 				if err != nil {
 					ThrowOttoException(&call, err.Error())
 				}
-				out, err := exec.Command(command, stringArgs...).Output()
+
+				cmd := exec.Command(command, stringArgs...)
+				var stdout bytes.Buffer
+				cmd.Stdout = &stdout
+				cmd.Stderr = &stdout
+
+				done := make(chan struct{})
+				go func() {
+					if err = cmd.Run(); err != nil {
+						log.Debug("Run %s %s error: %s", command, stringArgs, err)
+					}
+					close(done)
+				}()
+
+				select {
+				case interrupt := <-call.Otto.Interrupt:
+					log.Debug("Received otto interrupt in gohan_exec")
+					if cmd.Process != nil {
+						if err := cmd.Process.Kill(); err != nil {
+							log.Debug("Kill %s %s failed: %s", command, stringArgs, err)
+						}
+					}
+					interrupt()
+				case <-done:
+				}
+
 				resp := map[string]string{}
 				if err != nil {
 					resp["status"] = "error"
 					resp["output"] = err.Error()
 				} else {
 					resp["status"] = "success"
-					resp["output"] = string(out)
+					resp["output"] = stdout.String()
```

Comment:
In my view they do the same thing, see implementation https://golang.org/src/bytes/buffer.go?s=1909:1941#L38. Could you elaborate a bit more what exactly has changed?